backend/services/workflows/app/processor/string.go

```diff
@@ -136,6 +139,38 @@ SubMatchLoop:
 			// Replace ${workflow.input.KEY} with the KEY input variable
 			paramName := varName[len(workflowInputVariable):]
 			for _, param := range j.job.InputParameters {
+				// support for key === "device.jsonInput.tier"; accessing the input params
+				// objects by JSON identifiers. for instance, if a whole device is passed
+				// as an input parameter, then with `device.jsonInput.tier` you can refer
+				// to the Device.Tier field
+				if strings.HasPrefix(paramName, param.Name+jsonRefInputVariable) {
+					dataAny := gojsonq.New().FromString(param.Value).Find(
+						strings.TrimPrefix(
+							paramName,
+							param.Name+jsonRefInputVariable,
+						),
+					)
+					switch d := dataAny.(type) {
+					case string:
+						return d
+					case float64:
+						return fmt.Sprintf("%.8f", d)
+					case float32:
+						return fmt.Sprintf("%.8f", d)
+					case int:
+						return fmt.Sprintf("%d", d)
+					case int32:
+						return fmt.Sprintf("%d", d)
+					case int64:
+						return fmt.Sprintf("%d", d)
+					case bool:
+						if d {
+							return "true"
+						} else {
+							return "false"
+						}
+					}
```

Comment:
This snippet will not output anything if dataType is none of the types in this switch (e.g. `null`). How about using the json codec by default?
```suggestion
					switch d := dataAny.(type) {
					case string:
						return d
					default:
						b, _ := json.Marshal(dataAny)
						return string(b)
					}
```

Reply:
Or even just 
```suggestion
					return fmt.Sprint(dataAny) // may not be too nice with floats
```

Reply:
as you wish. (and I did want to only support primitive types) but it is ok.